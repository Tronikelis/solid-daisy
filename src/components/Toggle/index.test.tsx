/* eslint-disable solid/reactivity */
import { render } from "@solidjs/testing-library";
import userEvent from "@testing-library/user-event";
import { createSignal } from "solid-js";
import { describe, expect, it } from "vitest";

import { Toggle } from ".";

describe("<Toggle />", () => {
    it("renders correctly", () => {
        const { container } = render(() => <Toggle />);
        expect(container).toMatchSnapshot();
    });

    it("controlled input works", async () => {
        const [checked, setChecked] = createSignal(false);

        const screen = render(() => (
            <Toggle
                label="foo"
                data-testid="foo"
                checked={checked()}
                onInput={e => setChecked(e.target.checked)}
            />
        ));

        expect(screen.getByTestId("foo")).not.toBeChecked();

        await userEvent.click(screen.getByText("foo"));

        expect(screen.getByTestId("foo")).toBeChecked();
        expect(checked()).toBe(true);

        setChecked(false);
        expect(screen.getByTestId("foo")).not.toBeChecked();
    });
});
