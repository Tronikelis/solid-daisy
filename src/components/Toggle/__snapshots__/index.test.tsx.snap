// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Toggle /> > renders correctly 1`] = `
<div>
  <div
    class="flex flex-row items-center gap-4"
  >
    <input
      class="toggle"
      type="checkbox"
    />
  </div>
</div>
`;
